### PR TITLE
Add exit condition for repeat feature

### PR DESCRIPTION
**Pull request by renfeiw**
- exit when success is detected

Signed-off-by: renfeiw <renfeiw@ca.ibm.com>

### CODE REVIEW

**Comment by renfeiw** [comment]
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/5764

**Review by LongyuZhang — APPROVED** [review]
LGTM

File: settings.mk [hunk 1]
```diff
@@ -229,6 +229,9 @@ else
 	ifeq ($(EXIT_FAILURE), true)
 		EXIT_CONDITION+=|| $(HAS_FAILURE)
 	endif
+	ifeq ($(EXIT_SUCCESS), true)
+		EXIT_CONDITION+=|| [ $$success -ne 0 ]
```
Comment (karianna) [inline settings.mk:233]: Isn't this adding a success condition?  I'm confused :-)

Comment (renfeiw) [reply settings.mk:233]: This is part of the repeat feature: https://github.com/adoptium/TKG/issues/352. When we repeat the tests, we would like to have the option EXIT_SUCCESS to stop repeating when encountering a success in last iteration. This is useful for triaging purpose.

**Comment by renfeiw** [comment]
related: https://github.com/adoptium/TKG/issues/352

**Comment by renfeiw** [comment]
I have converted this PR to draft and wait util TKG is branched for Oct release.

**Comment by smlambert** [comment]
I have tagged TKG yesterday (v0.9.5), so we are fine to continue to work in the main branch, it won't affect the October release.

File: settings.mk (hunk 1 above)
Comment (smlambert) [reply settings.mk:233]: @karianna - there are cases where we want to prove a test can pass, so run for X iterations or until it passes (exit on first instance of it passing).  There are cases where we want to prove a test will fail, so run for X iterations or until if fails (exit on first instance of it failing), and finally cases where we want to run for X iterations and gather the pass/fail rate for the entire run (do not exit until X iterations are complete).

**Pull request merged**